lib/presentation/widget/shared_space_document/shared_space_document_viewmodel.dart

```diff
@@ -0,0 +1,846 @@
+import 'package:open_file/open_file.dart' as open_file;
+import 'package:permission_handler/permission_handler.dart';
+import 'package:redux/src/store.dart';
+import 'package:redux_thunk/redux_thunk.dart';
+import 'package:share/share.dart' as share_library;
+
+class SharedSpaceDocumentNodeViewModel extends BaseViewModel {
+  final AppNavigation _appNavigation;
+  final LocalFilePicker _localFilePicker;
+  final VerifyNameInteractor _verifyNameInteractor;
+  final GetAllChildNodesInteractor _getAllChildNodesInteractor;
+  final CreateSharedSpaceFolderInteractor _createSharedSpaceFolderInteractor;
+  final GetSorterInteractor _getSorterInteractor;
+  final SaveSorterInteractor _saveSorterInteractor;
+  final SortInteractor _sortInteractor;
+  final RenameSharedSpaceNodeInteractor _renameSharedSpaceNodeInteractor;
+  final SearchWorkGroupNodeInteractor _searchWorkGroupNodeInteractor;
+  final DownloadWorkGroupNodeInteractor _downloadWorkGroupNodeInteractor;
+  final DownloadPreviewWorkGroupDocumentInteractor _downloadPreviewWorkGroupDocumentInteractor;
+  final DownloadMultipleNodeIOSInteractor _downloadMultipleNodeIOSInteractor;
+  final CopyMultipleFilesToMySpaceInteractor _copyMultipleToMySpaceInteractor;
+  final CopyMultipleFilesToSharedSpaceInteractor _copyMultipleFilesToSharedSpaceInteractor;
+  final RemoveMultipleSharedSpaceNodesInteractor _removeMultipleSharedSpaceNodesInteractor;
+
+  StreamSubscription _storeStreamSubscription;
+
+  SearchQuery _searchQuery = SearchQuery('');
+
+  SearchQuery get searchQuery => _searchQuery;
+
+  List<WorkGroupNode> _workGroupNodesList;
+  SharedSpaceDocumentArguments _documentArguments;
+
+  SharedSpaceDocumentNodeViewModel(
+    Store<AppState> store,
+    this._appNavigation,
+    this._localFilePicker,
+    this._verifyNameInteractor,
+    this._getAllChildNodesInteractor,
+    this._createSharedSpaceFolderInteractor,
+    this._getSorterInteractor,
+    this._saveSorterInteractor,
+    this._sortInteractor,
+    this._renameSharedSpaceNodeInteractor,
+    this._searchWorkGroupNodeInteractor,
+    this._downloadWorkGroupNodeInteractor,
+    this._downloadPreviewWorkGroupDocumentInteractor,
+    this._downloadMultipleNodeIOSInteractor,
+    this._copyMultipleToMySpaceInteractor,
+    this._copyMultipleFilesToSharedSpaceInteractor,
+    this._removeMultipleSharedSpaceNodesInteractor,
+  ) : super(store) {
+    _storeStreamSubscription = store.onChange.listen((event) {
+      event.sharedSpaceState.viewState.fold((failure) => null, (success) {
+        if (success is SearchWorkGroupNodeNewQuery && event.uiState.searchState.searchStatus == SearchStatus.ACTIVE) {
+          _search(success.searchQuery);
+        } else if (success is DisableSearchViewState) {
+          store.dispatch(SharedSpaceDocumentSetSearchResultAction(_workGroupNodesList));
+          _searchQuery = SearchQuery('');
+        }
+      });
+    });
+  }
```

Comment:
can we have a better way for that?

Reply:
Because the search screen is shared at the home screen. So here I create a stream to listen for the changes from that search menu. Furthermore the app is syncing this way. I think that's the single most solvable method we can use right now. WDYT.

Reply:
yes, since search bar at Home Screen, its the best solution we have so far.